.github/workflows/test_esp_app.yml

```diff
@@ -0,0 +1,58 @@
+name: Run Test Application
+
+on:
+  workflow_call: {}
+  workflow_dispatch: {}
+jobs:
+  run-target:
+    name: Run Test App on target
+    runs-on: [self-hosted, linux, docker, "${{ matrix.espidf_target }}"]  # (expects tags with the same name as espidf_target)
+    container:
+      image: python:3.7-buster
+      options: --privileged  # (privileged mode has access to serial ports)
+    strategy:
+      fail-fast: false
+      matrix:
+        espidf_target: ["esp32", "esp32c3", "esp32s2", "esp32s3"]  # <--- Specify a list of Espressif chips connected to your runner
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Download files / artifacts from build job
+        uses: actions/download-artifact@v2
+        with:
+          name: test_app_bin_${{ matrix.espidf_target }}
+          path: test_app/build
+      - name: Install Python packages for PyTest
+        env:
+          PIP_EXTRA_INDEX_URL: "https://www.piwheels.org/simple"
```

Comment:
@pedrominatel Please replace by link to Espressif Python wheels registry:
```yml
   PIP_EXTRA_INDEX_URL: "https://dl.espressif.com/pypi"
```

This should fix failing tests in CI.